### ISSUE TITLE
App will not connect to DB perhaps because of ENV/conf vars

### ISSUE THREAD

**Issue by alternegro** [heroku/sbt-heroku#13]
After deploying with the plugin, my app crashes on startup because it cannot connect to the postgres database. My app reads the db connection info from an environment specific config file that is set in the Procfile with "-Dconfig.resource=${PLAY_CONFIG_FILE}"
Is there a step I'm missing that prevents this setup that worked using the git deploy from working with the deployPlugin?

**Comment by 1indsay** [comment]
In the documentation :

> You may set process types (similar to a Procfile) with herokuProcessTypes:

```
herokuProcessTypes in Compile := Map(
  "web" -> "target/universal/stage/bin/my-app -Dhttp.port=$PORT",
  "worker" -> "java -jar target/universal/stage/lib/my-worker.jar"
)
```

You can add your resource this way:
```
"web" -> "target/universal/stage/bin/my-app -Dhttp.port=$PORT -Dconfig.resource=$PLAY_CONFIG_FILE"
```

**Comment by jkutner** [comment]
You can define process types in the `build.sbt` as @1indsay points out. But as long as you are using sbt-heroku version 0.3.x, then it should pick up your Procfile too. 

What is the path of the value set for `PLAY_CONFIG_FILE`? If it's not under `target/universal/stage/` you will need to include it in the slug like this:

```scala
herokuIncludePaths in Compile := Seq("conf")
```

**Comment by alternegro** [comment]
Thanks@1indsay & @jkutner .
I'm using 0.3.3 so I think adding the process types in build.sb will be redundant but I'll give that a shot anyway. The config property resolves to *-Dconfig.resource=staging.conf* and I do have target/universal/stage/conf/staging.conf

**Comment by alternegro** [comment]
Setting *herokuProcessTypes* in build.sbt made the difference. So running 0.3.3 didn't pick up the Procfile automagically.

**Comment by jkutner** [comment]
Ah, that's a bug. I've fixed it in version 0.3.4. Thanks for letting us know!
